addon/wait-for-promise.ts

```diff
@@ -1,10 +1,11 @@
-import { DEBUG } from '@glimmer/env';
 import TestWaiter from './test-waiter';
 
 const PROMISE_WAITER = new TestWaiter<Promise<unknown>>('promise-waiter');
 
 /**
- * A convenient utility function to simplify waiting for a promise.
+ * A convenient utility function to simplify waiting for a promise. Can be used
+ * in both decorator and function form, and the function form either accepts a
+ * promise or an async function.
  *
  * @public
  * @param promise {Promise<T>} The promise to track async operations for
```

Comment:
Can we update the jsdoc to reflect the new signature?

Reply:
I'm actually very lost as far as the documentation goes typescript + jsdoc + overloaded function signatures = a very confused Ben, and googling hasn't really helped. I think I'm going to need someone that understands it all to help out here.